Assets/Scripts/CoinManagerScript.cs

```diff
@@ -0,0 +1,105 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using DG.Tweening;
+
+public class CoinManagerScript : MonoBehaviour
+{
+
+    [SerializeField] GameObject coinPrefab;
+    [SerializeField] Transform coinTarget;
+    [SerializeField] Text textField;
+
+    [Space]
+    [Header("Available coins: (coins to pool)")]
+    [SerializeField] int maxCoins;
+
+    [Space]
+    [Header("Animation settings:")]
+    [SerializeField] [Range(0.5f, 2.9f)] float minAnimationDuration;
+    [SerializeField] [Range(2.9f, 0.5f)] float maxAnimationDuration;
+    [SerializeField] Ease easeType;
+    [SerializeField] float spread;
```

Comment:
This is neat, to avoid putting to many public variables and to organize it better in the inspector. Will do this going forward!